### PR TITLE
docs: Migrate repo documentation to a readthedocs.io wiki

### PR DESCRIPTION
**Pull request by spencer-tb**
This PR will start the process of creating a wiki/documentation page for this repo. 

Much like https://ethereum-tests.readthedocs.io/en/latest/#.

Upon approval of this PR, the current README.md documentation will be available here. 
https://execution-spec-tests.readthedocs.io/en/latest/

Extra documentation will be made in a separate PR.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Configuration file for the Sphinx documentation builder.
+#
+# For the full list of built-in configuration values, see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = 'Ethereum Execution Spec Tests'
```
Comment (marioevz) [inline docs/conf.py:9]: Can we change these values to be similar to this:
https://github.com/ethereum/tests/blob/69c4c2a98dc8a712d4c6f5a817da4f21ff719006/docs/conf.py#L48-L51

**Review by marioevz — CHANGES_REQUESTED** [review]
Just a small comment, but overall looks pretty good :)

File: docs/conf.py (hunk 1 above)
Comment (spencer-tb) [reply docs/conf.py:9]: Yes definitely! Apologies - they where generated automatically through the command line :)

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+A new set of tests for ethereum execution clients to test against. The [git repo](https://github.com/ethereum/execution-spec-tests) is updated regulary with new tests.
+
+## Getting Started
+- ###[Quick Start](getting_started/01_quick_start.md)
+- ###[Package Overview](getting_started/02_package_overview.md)
+- ###[Writing Tests](getting_started/03_writing_tests.md)
+
+## Tutorials
+- ###[Introduction](tuorials/01_)
```
Comment (marioevz) [inline docs/index.md:16]:
```suggestion
<h2>Getting Started</h2>
- <h3>[Quick Start](getting_started/01_quick_start.md)</h3>
- <h3>[Package Overview](getting_started/02_package_overview.md)</h3>
- <h3>[Writing Tests](getting_started/03_writing_tests.md)</h3>
<h2>Tutorials</h2>
- <h3>[Introduction](tutorials/01_state_transition)</h3>
```
This removes the permalinks within this page, which looked a bit confusing, and also removes themo from the navigation.

**Review by marioevz — CHANGES_REQUESTED** [review]
Just a couple of minor comments but otherwise looks great!

**Comment by spencer-tb** [comment]
Once merged the tutorial for state tests will not show on the docs until #55 is merged.

File: docs/getting_started/01_quick_start.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+The following are required to either generate or develop tests:
+
+1. Python `3.10.0`.
+2. [`go-ethereum`](https://github.com/ethereum/go-ethereum) `geth`'s `evm` utility must be accessible in the `PATH`, typically at the latest version. To get it:
+     1. Install [the Go programming language](https://go.dev/doc/install) on your computer.
+     2. Clone [the Geth repository](https://github.com/ethereum/go-ethereum).
+     3. Run `make all`.
+     4. Copy `build/bin/evm` to a directory on the path.
+   
+   **Note:** To update to a different Geth branch (for example one that supports a specific EIP) all you need to do is to change the `evm` in the path.
```
Comment (marioevz) [inline docs/getting_started/01_quick_start.md:14]:
```suggestion
    **Note:** To update to a different Geth branch (for example one that supports a specific EIP) all you need to do is to change the `evm` in the path.
```

**Review by marioevz — APPROVED** [review]
LGTM

**Pull request merged**